vertical-pod-autoscaler/initializer/initializer.go

```diff
@@ -0,0 +1,347 @@
+
+	"github.com/golang/glog"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/apimock"
+	recommender "k8s.io/autoscaler/vertical-pod-autoscaler/recommender_mock"
+)
+
+const (
+	vpaInitializerConfigName = "vpa-initializer-config.k8s.io"
+	vpaInitializerName       = "vpa-initializer.k8s.io"
+)
+
+// Initializer initializes all pods matched by an existing vpa object
+// by assigning them recommended resource requests.
+type Initializer interface {
+	Run(stopCh <-chan struct{})
+}
+
+type initializer struct {
+	client          kubeclient.Interface
+	vpaLister       apimock.VerticalPodAutoscalerLister
+	podListerSynced cache.InformerSynced
+	informer        cache.SharedInformer
+	recommender     recommender.CachingRecommender
+	registerer      admissionregistrationv1alpha1.InitializerConfigurationInterface
+}
+
+func (i *initializer) Run(stopCh <-chan struct{}) {
+	glog.Infof("Starting VPA initializer.")
+	defer glog.Infof("Shutting down VPA initializer.")
+
+	i.register()
+	defer i.unregister()
+	i.informer.Run(stopCh)
+
+	if !cache.WaitForCacheSync(stopCh, i.podListerSynced) {
+		glog.Error("Cache sync failed.")
+		return
+	}
+}
+
+func (i *initializer) register() error {
+	configuration := newConfiguration()
+	config, err := i.registerer.Create(configuration)
+	if err != nil {
+		//TODO: If exists, return nil
+		glog.Infof("Error registering VPA Initializer Config: %v", err)
+		return err
+	}
+	glog.V(4).Infof("Registered config: %+v", config)
+	return nil
+}
+
+func (i *initializer) unregister() {
+	i.registerer.Delete(vpaInitializerConfigName, &metav1.DeleteOptions{})
+}
+
+func (i *initializer) updateResourceRequests(obj interface{}) {
+	pod := obj.(*v1.Pod)
+
+	if !shouldInitialize(pod) {
+		glog.V(4).Infof("Not updating pod: %v", pod.Name)
+		return
+	}
+	initializedPod, err := i.getInitializedPod(pod)
+
+	if err == nil {
+		// TODO(bskiba): Right now this update always fails if we try to change resource requests
+		// as there are checks in place that prevent from changing requests of a created pod.
+		result, err := i.client.CoreV1().Pods(pod.Namespace).Update(initializedPod)
+		if err == nil {
+			glog.V(3).Infof("Successfully updated pod %+v", result)
+			return
+		}
+	}
+	failedPod, err := getFailedPod(pod)
+	if err == nil {
+		i.client.CoreV1().Pods(pod.Namespace).Update(failedPod)
+	} else {
+		log.Fatalf("Unable to mark failed initialization for pod %v: %v", pod.Name, err)
+	}
+
+}
+
+func getFailedPod(pod *v1.Pod) (*v1.Pod, error) {
+	failedPodCopy, err := api.Scheme.Copy(pod)
+	if err != nil {
+		return nil, err
+	}
+	failedPod := failedPodCopy.(*v1.Pod)
+	markInitializationFailure(failedPod)
+	return failedPod, nil
+}
+
+// getInitializedPod returns a pod in an initialized state, updating resource
+// requests if applicable.
+func (i *initializer) getInitializedPod(pod *v1.Pod) (*v1.Pod, error) {
+
+	glog.V(4).Infof("Updating requirements for pod: %v", pod.Name)
```

Comment:
Add before/after or point me to the place where it is printed elsewhere.

Reply:
This is printed for every updated container and resource in line 225/229